### ISSUE TITLE
Markdown Math is printed twice

### ISSUE THREAD

**Issue by maxZangl** [PDConSec/vsc-print#266]
When printing math expressions, the extension adds the expression twice,   
once translated to math and once without translation to math

e.g. 

the single line:

`$S_0 * M_n = 7^2$`

will look like this in the pdf:

![image](https://github.com/PDConSec/vsc-print/assets/74556147/8edeafb4-d3ad-40c7-9e7c-73c6219144e8)

**Comment by PeterWone** [comment]
See https://github.com/PDConSec/vsc-print/discussions/258

**Comment by PeterWone** [comment]
@maxZangl Was your issue the same problem as #258? 

If not please share any additional information. Otherwise I'll have to treat this as a duplicate and close it.

**Comment by PeterWone** [comment]
I'm going to close this. If you object because adding a stylesheet didn't resolve it for you then add details and I'll re-open it.

**Comment by maxZangl** [comment]
Hi,

so adding the same stylesheet as in [#258](https://github.com/PDConSec/vsc-print/discussions/258) is giving me the following result:
![image](https://github.com/PDConSec/vsc-print/assets/74556147/0757fe00-7c36-436c-a997-82c7e226bca6)

This does solve the problem of duplicate math, however it is not usable.
I tried other stylesheets, like this https://github.com/sindresorhus/github-markdown-css/blob/main/github-markdown.css
With the following result:
![image](https://github.com/PDConSec/vsc-print/assets/74556147/beb9e16e-70c1-476b-8d13-1f7a9cfaccec)  
The markdown in VS Code itself is changed though, by using this second stylesheet:   
![image](https://github.com/PDConSec/vsc-print/assets/74556147/a5817301-fc3c-4fd7-9962-68314b29449e)


Also no matter what i do, the color scheme settings are not doing anything. I selected a few.  

I do not know if the bug above is therefore fixed or not.  

But i could not get the extension to print a single line of readable math yet unfortunately. I would like to though.

**Comment by PeterWone** [comment]
Max, what's rendering the LaTeX for you? Dump your installed extensions like so

```
code --list-extensions
```

and share the result. You can shorten the list if you want to edit out extensions you know are irrelevant to Markdown. For example, my system has 30 or so but only the following are relevant.
```
bierner.markdown-checkbox
bierner.markdown-mermaid
DavidAnson.vscode-markdownlint
fcrespo82.markdown-table-formatter
goessner.mdmath
mathpix.vscode-mathpix-markdown
pdconsec.vscode-print
tomoyukim.vscode-mermaid-editor
```
I should warn you that recently the person who maintains the built-in Markdown rendering pipeline has started to add in Markdown enhancements. He has an utter disregard for Print and frequently introduces breaking changes. If you didn't install a separate markdown extension like 

- `goessner.mdmath`
- `mathpix.vscode-mathpix-markdown`

this may be the source of your problems. Try installing one of these.

**Comment by maxZangl** [comment]
This is mycode --list-extensions output:

```
bierner.markdown-image-size
GitHub.codespaces
goessner.mdmath
koehlma.markdown-math
mathpix.vscode-mathpix-markdown
mlxprs.mlxprs
ms-python.isort
ms-python.python
ms-python.vscode-pylance
ms-toolsai.jupyter
ms-toolsai.jupyter-keymap
ms-toolsai.jupyter-renderers
ms-toolsai.vscode-jupyter-cell-tags
ms-toolsai.vscode-jupyter-slideshow
ms-vscode-remote.remote-ssh
ms-vscode-remote.remote-ssh-edit
ms-vscode.remote-explorer
mushan.vscode-paste-image
pdconsec.vscode-print
yzane.markdown-pdf
yzane.markdown-pdf`
```

Also i tested with disabling all extensions but:
```
goessner.mdmath
pdconsec.vscode-print
```
And also tried only:
```
mathpix.vscode-mathpix-markdown
pdconsec.vscode-print
```
And also tried all but:
```
pdconsec.vscode-print
```
In all **three** cases the math expression is printed twice..
I checked all settings related to markdowns but did not find anything i should change.

thank you for the warning and your help.

**Comment by PeterWone** [comment]
This problem occurs on my work notebook but not on my personal computer. I'm still trying to figure out the difference but it's probably environmental.

**Comment by PeterWone** [comment]
False alarm, I forgot that changing settings in my dev copy of vscode affect a running extension debug host. 

Using the goessner extension, having no extra stylesheets like this

![image](https://github.com/PDConSec/vsc-print/assets/5498936/bcaa9392-ef88-4dbb-977c-b0b2012d366e)

produces this

![image](https://github.com/PDConSec/vsc-print/assets/5498936/dcae1bfd-ff44-4678-96d3-2a6b4cef093a)

Adding this

![image](https://github.com/PDConSec/vsc-print/assets/5498936/6669770f-b7a8-488f-b968-bea198f18156)

produces this

![image](https://github.com/PDConSec/vsc-print/assets/5498936/c37811e7-ed5d-4b23-9374-31af4204b978)

**Comment by maxZangl** [comment]
yeah i just tested it again with the same stylesheet and it works fine now..
That is great thank you!

however i tested the same stylesheet earlier and it messed up the math expressions.... 
I just found the culprit, using this extension: https://marketplace.visualstudio.com/items?itemName=koehlma.markdown-math
is messing up the math, as in my second comment. That is good to know.

but anyway i am happy
Thanks for your help :D nice extension! I will use it frequently now

**Comment by PeterWone** [comment]
Max - historically I have tried to use VS Code's Markdown rendering pipeline in order to keep Print in sync with the Markdown preview pane. But Microsoft decisions are making this problematic, and I will probably instead support my own pipeline. Once you can serve the rendered page and its resource to a browser there is fundamentally no need for a built it preview. Having it in a browser also lets you put the preview on a second monitor, something VS Code cannot support directly.

When I do that, I'll probably take a bunch of the most useful markdown extensions, bake them into Print and resolve the resource and compatibility issues. To that end I need to know what people are using. I could rip a list of markdown extensions out of people's installations and send it to my server but some people would get bent out of shape over privacy (I don't care _who_ has what, just what and how many). 

So I'm interested in how people are using it. Tell me what you do with it, if you want to influence these decisions.

**Comment by maxZangl** [comment]
Hi Peter,

i am using markdowns as my preferred format to document anything, but mostly:
- my lecture notes, where i need to write math expressions and copy/paste images of definitions / examples
- small pieces of documentation of a project (for myself) where i add code sections and paste images
 
So i always need to be able to paste images very easily, where currently i use a combination of the tool _greenshot_ and this extension: https://marketplace.visualstudio.com/items?itemName=mushan.vscode-paste-image to paste the results.

Why do i need the print function from your extension?
To convert these markdowns into pdfs which i can easily read on my phone or share, since there is no good app to display markdowns in the same way as i have them in VS Code.

So basically i use markdowns as offline single confluence pages with decent math expressions support.
And as long as the images are in the output and the math expressions are still readable i am a happy person.

Here is an **example** from my lecture notes, where i pasted some screenshot fom wikipedia and added my personal notes below, which use some math notation:

![image](https://github.com/PDConSec/vsc-print/assets/74556147/a0a89491-adb6-4dc8-a4be-84aa1d7256a0)
 ![image](https://github.com/PDConSec/vsc-print/assets/74556147/fcde79cf-f6a8-49d5-b9a9-eac40b389a2c)
